code/modules/shuttles/landmarks.dm

```diff
@@ -122,6 +124,43 @@
 		if(T.density)
 			T.ChangeTurf(get_base_turf_by_area(T))
 
+//Used for custom landing locations. Self deletes after a shuttle leaves.
+/obj/effect/shuttle_landmark/temporary
+	name = "Landing Point"
+	landmark_tag = "landing"
+	flags = SLANDMARK_FLAG_AUTOSET
+	var/timer_id
+
+/obj/effect/shuttle_landmark/temporary/Initialize(var/mapload, var/shuttle_name)
+	shuttle_restricted = shuttle_name
+	. = ..(mapload)
+
+/obj/effect/shuttle_landmark/temporary/Initialize()
+	landmark_tag += "-[random_id("landmarks",1,9999)]"
+	timer_id = addtimer(CALLBACK(src, .proc/check_for_shuttle), 1 MINUTE, TIMER_STOPPABLE | TIMER_LOOP)
```

Comment:
I feel it would be more graceful to avoid the checks entirely; delete if our shuttle leaves or if it clears its destination from this landmark (add a hook) without being in transit to here or here already. If you do want to keep it, seems simpler to `START_PROCESSING(SSprocessing, src)` instead.